suite-native/biometrics/src/useBiometrics.tsx

```diff
@@ -39,23 +42,44 @@ export const useBiometrics = () => {
             return;
         }
 
-        if (isBiometricsOptionEnabled && !isUserAuthenticated) {
+        if (
+            isBiometricsOptionEnabled &&
+            !isUserAuthenticated &&
+            !isBiometricsAuthenticationCanceled
+        ) {
             const result = await authenticate();
 
             if (result && result?.success) {
                 setIsUserAuthenticated(true);
                 setIsBiometricsOverlayVisible(false);
+
+                // If the user cancels the authentication by button, we need to ask him again later.
+                // Otherwise a lot of tries would render this feauture unusable blocking user from the app.
+                // see https://github.com/trezor/trezor-suite/issues/10647
+            } else if (result?.error === 'user_cancel') {
+                setIsBiometricsAuthenticationCanceled(true);
             } else {
                 handleAuthentication();
             }
         }
     }, [
         isBiometricsOptionEnabled,
         isUserAuthenticated,
+        isBiometricsAuthenticationCanceled,
         setIsUserAuthenticated,
         setIsBiometricsOverlayVisible,
+        setIsBiometricsAuthenticationCanceled,
     ]);
 
+    // This is to relaunch authentication if the user canceled it previously by a button
+    // and now wanted to reenable it.
+    useEffect(() => {
+        if (!isBiometricsAuthenticationCanceled) {
+            handleAuthentication();
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Please explain why this is necessary and why `handleAuthentication` is not included in deps list? If there is no valid reason, I'd prefer not to disable this rule.

Reply:
492763e1d2